### PR TITLE
API4: add SQL Function DAYOFWEEK

### PR DESCRIPTION
**Pull request by aydun**
Overview
----------------------------------------
Adds day of week functions as SQL transformation.   Let's you do things like 'find events on Wednesdays'

Before
----------------------------------------
No day of week functions

After
----------------------------------------
Day of week functions. 

Technical Details
----------------------------------------
The SQL function returns 1-7, mapped to day names by the 'getOptions()' function.

Comments
----------------------------------------
Has test!

### CODE REVIEW

**Comment by colemanw** [comment]
Hey @aydun this looks useful, thanks!
My one pushback would be to add 1 function instead of 2. The presence of the `getOptions()` means that numeric values will be returned by the API but will be displayed in SK as the name of the day. See `SqlFunctionMONTH` which follows that pattern.

File: tests/phpunit/api/v4/Action/SqlFunctionTest.php [hunk 1]
```diff
@@ -234,6 +234,8 @@ public function testDateFunctions() {
       ->addSelect('QUARTER(birth_date) AS quarter')
       ->addSelect('MONTH(birth_date) AS month')
       ->addSelect('EXTRACT(YEAR_MONTH FROM birth_date) AS year_month')
+      ->addSelect('DAYOFWEEK(birth_date) AS day_number')
+      ->addSelect('DAYNAME(birth_date) AS day_name')
```
Comment (colemanw) [inline tests/phpunit/api/v4/Action/SqlFunctionTest.php:238]:
```suggestion
      ->addSelect('DAYOFWEEK(birth_date):label AS day_name')
```
Should do the same thing right?

Comment (aydun) [reply tests/phpunit/api/v4/Action/SqlFunctionTest.php:238]: Hmm - that produces:
```
There was 1 error:

1) api\v4\Action\SqlFunctionTest::testDateFunctions
CRM_Core_Exception: Aliasing field names is not allowed, only expressions can have an alias.
```

**Comment by aydun** [comment]
Hi @colemanw I agree it would be better just to have the one function, but how can I get the day name to show in SK results?  The names show up in filters and where clauses, but not search results

**Comment by colemanw** [comment]
You're right @aydun that was missing. This fixes it https://github.com/civicrm/civicrm-core/pull/26065

**Comment by aydun** [comment]
Hey @colemanw, that's great - I've updated this just to use the one function.    Tests won't pass until #26065 is merged.

**Comment by colemanw** [comment]
retest this please

**Comment by colemanw** [comment]
@aydun I think this needs a `git pull --rebase origin master`

**Comment by aydun** [comment]
OK, let's see if that gets the tests to pass....

**Comment by seamuslee001** [comment]
ok this seems to be passing tests now and Coleman has reviewed this so seems fine to me

**Comment by aydun** [comment]
Thanks @seamuslee001

**Pull request merged**